### PR TITLE
ncm-postgresql: rewrite (and support el7)

### PR DESCRIPTION
**Pull request by stdweird**
full rewrite, but same logic as old component

### CODE REVIEW

**Comment by stdweird** [comment]
retest this please

File: ncm-postgresql/src/main/pan/components/postgresql/schema.pan [hunk 1]
```diff
@@ -2,35 +2,27 @@
...
+include 'quattor/types/component';
 
-function pgsql_is_hba_db = {
+function postgresql_is_hba_db = {
     # Check cardinality and type of argument.
     if (ARGC != 1 || !is_string(ARGV[0]))
         error("usage: is_asndate(string)");
-    if (match(ARGV[0],"all|sameuser|samerole|replication")) {
+
+    if (match(ARGV[0], "all|sameuser|samerole|replication")) {
```
Comment (ned21) [inline ncm-postgresql/src/main/pan/components/postgresql/schema.pan]: Please anchor the regexp.

File: ncm-postgresql/src/main/pan/components/postgresql/schema.pan [hunk 2]
```diff
@@ -2,35 +2,27 @@
...
         return(true);
```
Comment (ned21) [inline ncm-postgresql/src/main/pan/components/postgresql/schema.pan]: Drop the return()s for more efficient PAN.

File: ncm-postgresql/src/main/pan/components/postgresql/schema.pan [hunk 3]
```diff
@@ -2,35 +2,27 @@
...
     };
 };
 
-function pgsql_is_hba_address = {
+function postgresql_is_hba_address = {
     # Check cardinality and type of argument.
     if (ARGC != 1 || !is_string(ARGV[0]))
         error("usage: is_asndate(string)");
+
     if (match(ARGV[0],"samehost|samenet")) {
```
Comment (ned21) [inline ncm-postgresql/src/main/pan/components/postgresql/schema.pan]: Anchor regexes

**Comment by ned21** [comment]
Github is coping well with commit 6e335c2 and is not showing it in the diff viewer. Looks like NoAction support is missing - please can you add it? (Should we have a unit test that checks this is enabled for all new components and rewrites?)

**Comment by stdweird** [comment]
now based on #654

File: ncm-postgresql/src/main/pan/components/postgresql/schema.pan [hunk 4]
```diff
@@ -2,37 +2,29 @@
...
 declaration template components/postgresql/schema;
 
-include {'quattor/schema'};
+include 'quattor/types/component';
 
-function pgsql_is_hba_db = {
+function postgresql_is_hba_db = {
     # Check cardinality and type of argument.
     if (ARGC != 1 || !is_string(ARGV[0]))
         error("usage: is_asndate(string)");
```
Comment (ned21) [inline ncm-postgresql/src/main/pan/components/postgresql/schema.pan]: Error message has incorrect function name. (Use FUNCTION global to guarantee correctness?)

File: ncm-postgresql/src/main/pan/components/postgresql/schema.pan [hunk 5]
```diff
@@ -2,37 +2,29 @@
...
-        return(exists("/software/components/postgresql/databases/" + ARGV[0]));
+        exists("/software/components/postgresql/databases/" + ARGV[0]);
     };
 };
 
-function pgsql_is_hba_address = {
+function postgresql_is_hba_address = {
     # Check cardinality and type of argument.
     if (ARGC != 1 || !is_string(ARGV[0]))
         error("usage: is_asndate(string)");
```
Comment (ned21) [inline ncm-postgresql/src/main/pan/components/postgresql/schema.pan]: ditto

**Comment by ned21** [comment]
Minor remarks but otherwise I think it's good to go once the predecessor PR is merged.

**Comment by stdweird** [comment]
@ned21 i didn't know about `FUNCTION`.

**Pull request merged**